src/MessageHandler/ZipFileHandler.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace App\MessageHandler;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+
+use App\Entity\DatasetSubmission;
+
+use App\Message\ZipFile;
+
+use App\Repository\FileRepository;
+
+use App\Util\Datastore;
+use App\Util\ZipFiles;
+
+/**
+ * Handler for zipping files for download.
+ */
+class ZipFileHandler implements MessageHandlerInterface
+{
+    /**
+     * The monolog logger.
+     *
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * Zip files utility class.
+     *
+     * @var ZipFiles
+     */
+    private $zipFiles;
+
+    /**
+     * Download directory for the zip file.
+     *
+     * @var string
+     */
+    private $downloadDirectory;
+
+    /**
+     * The File Repository.
+     *
+     * @var FileRepository
+     */
+    private $fileRepository;
+
+    /**
+     * Entity Manager interface instance.
+     *
+     * @var EntityManagerInterface
+     */
+    private $entityManager;
+
+    /**
+     * Datastore utility instance.
+     *
+     * @var Datastore
+     */
+    private $datastore;
+
+    /**
+     * ZipFileHandler constructor.
+     *
+     * @param LoggerInterface        $logger            Default Monolog logger interface.
+     * @param ZipFiles               $zipFiles          Zip files utility instance.
+     * @param string                 $downloadDirectory Temporary download directory path.
+     * @param FileRepository         $fileRepository    File repository to query objects.
+     * @param EntityManagerInterface $entityManager     Entity manager interface instance.
+     * @param Datastore              $datastore         Datastore utility instance.
+     */
+    public function __construct(
+        LoggerInterface $logger,
+        ZipFiles $zipFiles,
+        string $downloadDirectory,
+        FileRepository $fileRepository,
+        EntityManagerInterface $entityManager,
+        Datastore $datastore
+    ) {
+        $this->logger = $logger;
+        $this->zipFiles = $zipFiles;
+        $this->downloadDirectory = $downloadDirectory;
+        $this->fileRepository = $fileRepository;
+        $this->entityManager = $entityManager;
+        $this->datastore = $datastore;
+    }
+
+    /**
+     * Invoke function to zip files for download.
+     *
+     * @param ZipFile $zipFile
+     */
+    public function __invoke(ZipFile $zipFile)
+    {
+        $this->logger->info(sprintf('Zipping files started'));
```

Comment:
Don't really need sprintf if you are not passing in stuff. Also I learned about a new magic syntax in logger:
`$context['something'] = "bla";
`$this->logger->info('Hello {something} , how are you?', $context);`

Reply:
Copy pasted, Yeah will remove it 👍